A77-xds-rate-limiting-rlqs.md

```diff
@@ -0,0 +1,618 @@
+for gRPC.
+
+| Attribute           | Type                  | gRPC source                  | Envoy Description                                           |
+|---------------------|-----------------------|------------------------------|-------------------------------------------------------------|
+| `request.path`      | `string`              | Full method name<sup>1</sup> | The path portion of the URL.                                |
+| `request.url_path`  | `string`              | Same as `request.path`       | The path portion of the URL without the query string.       |
+| `request.host`      | `string`              | Authority<sup>2</sup>        | The host portion of the URL.                                |
+| `request.scheme`    | `string`              | Not set                      | The scheme portion of the URL.                              |
+| `request.method`    | `string`              | `POST`<sup>3</sup>           | Request method.                                             |
+| `request.headers`   | `map<string, string>` | `metadata`<sup>4</sup>       | All request headers indexed by the lower-cased header name. |
+| `request.referer`   | `string`              | `metadata["referer"]`        | Referer request header.                                     |
+| `request.useragent` | `string`              | `metadata["user-agent"]`     | User agent request header.                                  |
+| `request.time`      | `timestamp`           | Not set                      | Time of the first byte received.                            |
+| `request.id`        | `string`              | `metadata["x-request-id"]`   | Request ID corresponding to `x-request-id` header value     |
+| `request.protocol`  | `string`              | Not set                      | Request protocol.                                           |
+| `request.query`     | `string`              | `""`                         | The query portion of the URL.                               |
+
+###### Footnotes
+
+**<sup>1</sup> `request.path`**
+
+* CPP: `metadata[":path"]`
+* Go: `grpc.Method(ctx)`
+* Java: `"/" + serverCall.getMethodDescriptor().getFullMethodName()`
+
+**<sup>2</sup> `request.host`**
+
+* CPP, Go: `metadata[":authority"]`
+* Java: `serverCall.getAuthority()`
+
+**<sup>3</sup> `request.method`**\
+Hard-coded to `"POST"` if unavailable and a code audit confirms the server
+denies requests for all other method types.
+
+**<sup>4</sup> `request.headers`**\
+As defined in [gRFC A41], "header" field.
+
+##### Implementation
+
+For performance reasons, CEL variables should be resolved on demand. CEL Runtime
+provides the different variable resolving approaches based on the language:
+
+* CPP: [`BaseActivation::FindValue()`](https://github.com/google/cel-cpp/blob/9310c4910e598362695930f0e11b7f278f714755/eval/public/base_activation.h#L35)
```

Comment:
I don't think we need to list the specific APIs in each language.

Reply:
Normally I'd agree, but it's surprisingly difficult to find and identify these - just because approaches are really different between languages. IMO it'll save the implementers some time. 

Do you think there's a way to keep these, but without making them look as the prescribed implementation? Maybe under a different section?